executor/hash_table.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package executor
+
+import (
+	"github.com/pingcap/tidb/util/chunk"
+)
+
+const maxEntrySliceLen = 8 * 1024
+
+type entry struct {
+	ptr  chunk.RowPtr
+	next entryAddr
+}
+
+type entryStore struct {
+	slices   [][]entry
+	sliceIdx uint32
+	sliceLen uint32
+}
+
+func (es *entryStore) put(e entry) entryAddr {
+	if es.sliceLen == maxEntrySliceLen {
+		es.slices = append(es.slices, make([]entry, 0, maxEntrySliceLen))
```

Comment:
I am a little worried that this is too big for oltp cases, have you tested it?

Reply:
I copied this line from `mvmap`, and I think it has a little problem.
In `newRowHashMap`, it makes a slice with 64 capacity but append a slice witch 8k capacity when first slice is full.

Since hash collision is not so frequent, it's not a big problem and I'll adjust it in further PR.

Reply:
How about other max capacities? For example, 32, 64, ...